nwastdlib/graphql/extensions/error_handler_extension.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2019-2023 SURF.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from collections.abc import Collection, Generator, Iterable
+from contextvars import ContextVar
+from enum import StrEnum, auto
+from http import HTTPStatus
+from typing import Any, Callable
+
+import structlog
+from graphql import GraphQLError
+from httpx import HTTPStatusError, Response
+from strawberry.extensions import SchemaExtension
+from strawberry.types import ExecutionContext, Info
```

Comment:
For example in the asyncio.py module we do

```
try:
    from anyio import CapacityLimiter, to_thread
except ImportError:
    warnings.warn("anyio is required for gather_nice_sync()", stacklevel=1)
else:

    async def gather_nice_sync(
```

Reply:
If you have to use these kind of solutions its better to have a separate package. For now I leave it like this since we are the main/only user of nwa-stdlib and we can change this if we run into issues.